### ISSUE TITLE
Design of Undo button

### ISSUE THREAD

**Issue by veillette** [phetsims/calculus-grapher#78]
Th design specifies that the undo button should have a back arrow.

![image](https://user-images.githubusercontent.com/7633344/197789072-840988f8-a070-438c-b393-5175923669bc.png)

I did a cursory search for the icon but I didn't find something similar in other sims. ( I found a straight to the left arrow for Undo buttons but nothing like this icon).
Maybe I'm missing it, but @amanda-phet, let me know if you are aware of other instances in PhET for this icon.
If not we can try  to find some font-awesome icon, or make a custom icon from the AI file.

**Comment by pixelzoom** [comment]
I'll give my usual spiel about icons.  It would be nice to use the fontawesome icon for undo (https://fontawesome.com/v4/icon/undo) instead of inventing and implementing something new/different.

**Comment by amanda-phet** [comment]
The Expression Exchange game uses this icon. 
![image](https://user-images.githubusercontent.com/7852916/197846302-88d428a8-8588-4711-b828-1edfd0019cde.png)

In general I always look to font awesome for universal icons too, but in this case I think we might not want to use that icon in particular. We already have three variants of an arrow in a circular shape (reset all, mini-reset, and refresh) I think it we probably shouldn't add a fourth.

**Comment by amanda-phet** [comment]
Build a Molecule uses a similar, but custom, button: 
![image](https://user-images.githubusercontent.com/7852916/197850500-640bb377-11b0-44b1-966a-23aa0f649f65.png)

**Comment by veillette** [comment]
For my own reference, the first example is the  UndoButton in Game in Expression-Exchange.  The shape is made through Kite. 

The second example is the Refill Button from common.view in build a molecule.
It makes use of a font awesome icon (Reply Solid Shape)
https://fontawesome.com/v5/icons/reply?s=solid&f=classic
The icon is rotated by 45 \degrees.

**Comment by veillette** [comment]
@pixelzoom , what is the best way to proceed? 

I could create my own rectangular pushbutton using the font awesome icon for Calculus Grapher but should this be raised to a common code issue?
and if so should we favor the kite implementation or the font awesome icon?

**Comment by pixelzoom** [comment]
If there's a fontawesome icon that matches what @amanda-phet has specified, I recommend using it.

If there's no fontawesome icon, then create scenery-phet `UndoIcon extends Path`.

In either case, there should be scenery-phet `UndoButton extends RectangularPushButton`, and it should be used in all sims that have this button. 

Let me know if this is something you'd like me to take on.

**Comment by pixelzoom** [comment]
Looking at expression-exchange UndoButton ... I would factor out UndoIcon, then move both UndoIcon and UndoButton to scenery-phet.

**Comment by pixelzoom** [comment]
Done in the above commits, and I created issues as needed in related sims (see links above). I also used UndoButton in calculus-grapher, deleted the unused string, and ran `grunt modulify`

@veillette please review UndoIcon, UndoButton, and the use of UndoButton in calculus-grapher.

**Comment by veillette** [comment]
@pixelzoom ,that looks good in calculus-grapher.

In scenery-phet/UndoButton, the options field sets the cursor to 'pointer' but the default inherited from the ButtonNode is already set to 'pointer'.  

In UndoIcon, the comments for the size options states that the icon is square. The comment is incorrect.  The height and width  of the icon shape are different. The size is merely there for scaling the icon, and we have a unique value because we want to preserve the aspect ratio.

**Comment by veillette** [comment]
I think the height is equal to size (for sure, the height >= size. ). It is a bit hard to parse the width and height of an icon with quadratic bezier curves.

**Comment by pixelzoom** [comment]
Thanks for the review.

In UndoButton, `cursor: 'pointer'` was in the implementation migrated from expression-exchange. I've removed it.

In UndoIcon, I changed `size` back to `height`.  It was `iconHeight` in the original expression-exchange implementation.

Back to @veillette for review, feel free to close.

**Comment by veillette** [comment]
THat looks good. Closing
